src/twisted/web/http.py

```diff
@@ -1205,6 +1206,16 @@ def write(self, data):
 
             for name, values in self.responseHeaders.getAllRawHeaders():
                 for value in values:
+                    if (
+                        self.method == b"CONNECT"
+                        and self.code // 100 == 2
+                        and name == b"Content-Length"
```

Comment:
The `name ==` comparison should be executed first so that the slightly more expensive retrieve-an-attribute / compare a string, whereas `name==` is just a local variable comparison which is slightly cheaper and we only care about doing this for one header.